frontend_tests/node_tests/filter.js

```diff
@@ -1572,3 +1572,39 @@ test("error_cases", (override) => {
     const predicate = get_predicate([["pm-with", "Joe@example.com"]]);
     assert.ok(!predicate({type: "private"}));
 });
+
+run_test("is_web_public_compatible", () => {
+    // tests same as test_is_web_public_compatible from test_message_fetch.py
+    assert.ok(Filter.is_web_public_compatible([]));
+    assert.ok(Filter.is_web_public_compatible([{operator: "has", operand: "attachment"}]));
+    assert.ok(Filter.is_web_public_compatible([{operator: "has", operand: "image"}]));
+    assert.ok(Filter.is_web_public_compatible([{operator: "search", operand: "magic"}]));
+    assert.ok(Filter.is_web_public_compatible([{operator: "near", operand: "15"}]));
+    assert.ok(
+        Filter.is_web_public_compatible([
+            {operator: "id", operand: "15"},
+            {operator: "has", operand: "attachment"},
+        ]),
+    );
+    assert.ok(Filter.is_web_public_compatible([{operator: "sender", operand: "hamlet@zulip.com"}]));
+    assert.ok(
+        !Filter.is_web_public_compatible([{operator: "pm-with", operand: "hamlet@zulip.com"}]),
+    );
+    assert.ok(
+        !Filter.is_web_public_compatible([
+            {operator: "group-pm-with", operand: "hamlet@zulip.com"},
+        ]),
+    );
+    assert.ok(Filter.is_web_public_compatible([{operator: "stream", operand: "Denmark"}]));
+    assert.ok(
+        Filter.is_web_public_compatible([
+            {operator: "stream", operand: "Denmark"},
+            {operator: "topic", operand: "logic"},
+        ]),
+    );
+    assert.ok(!Filter.is_web_public_compatible([{operator: "is", operand: "starred"}]));
+    assert.ok(!Filter.is_web_public_compatible([{operator: "is", operand: "private"}]));
+    assert.ok(Filter.is_web_public_compatible([{operator: "streams", operand: "public"}]));
+    // Malformed input not allowed
+    assert.ok(!Filter.is_web_public_compatible([{operator: "has"}]));
```

Comment:
This is OK, but I wonder if it'd be cleaned to do this in `test_basics` as an extra function to test for each of the cases we setup.

Reply:
For test, I am not sure if it makes sense to add 8-10 `if spectator` statements  in `Filter` functions, which we will forget to add if we add another one of these functions.

Reply:
Oh, I hadn't imagined we'd add any `if spectator` checks -- we'd just test `is_web_public_compatible` on the various Filter objects we setup as our standard set of test cases.

Reply:
I see, but yeah, that doesn't make sense if we're trying to expose this in `hash_util` as well.